### PR TITLE
style: Multiple calc serialization fixes.

### PR DESCRIPTION
**Pull request by emilio**
This puts us in line with the spec as written, except for caveat in https://github.com/w3c/csswg-drafts/issues/1731.

Need to fix computed calc() too, I'm less confident that we won't need to change test expectations.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18131)

### CODE REVIEW

**Comment by emilio** [comment]
cc https://github.com/w3c/csswg-drafts/issues/1731

File: components/style/values/computed/length.rs [hunk 1]
```diff
@@ -194,6 +194,18 @@ impl From<LengthOrPercentageOrNone> for Option<CalcLengthOrPercentage> {
...
+        // Also, serialisation here is kinda weird with negative percentages,
+        // for example, as of right now:
+        //
+        //  calc(10px - 5%)
+        //
+        // would serialize as:
+        //
+        //  calc(10px + -5%)
+        //
+        // Need to update and run through try.
```
Comment (canaltinova) [inline components/style/values/computed/length.rs]: I think we should change the computed value as well and test with a try run while we are at it. Is there a reason to prevents to do that right now?

**Review by canova — APPROVED** [review]
It looks good to me. I think we can change the computed value as well.
r=me

File: components/style/values/computed/length.rs (hunk 1 above)
Comment (emilio) [inline components/style/values/computed/length.rs]: The tree is closed.

Comment (emilio) [inline components/style/values/computed/length.rs]: (And I'd need to find a good revision, etc...)

I'll push a try run and if this hasn't landed tomorrow morning I'll just push it.

**Comment by emilio** [comment]
@bors-servo r=canaltinova

 * Will update the computed one ASAP.

File: components/style/values/computed/length.rs (hunk 1 above)
Comment (canaltinova) [inline components/style/values/computed/length.rs]: Well, I think this PR can wait, since it doesn't block anything, but either way is ok I guess.

**Comment by emilio** [comment]
@bors-servo r=canaltinova p=1 

 * The manifest :(

**Pull request merged**